docs/stdout-exporter-design.md

```diff
@@ -0,0 +1,223 @@
+# StdoutExporter Design
+
+In strongly typed languages typically there will be 2 separate `Exporter` interfaces, one that accepts spans from a tracer (SpanExporter) and one that accepts metrics (MetricsExporter)
+
+The exporter SHOULD be called with a checkpoint of finished (possibly dimensionally reduced) export records. Most configuration decisions have been made before the exporter is invoked, including which instruments are enabled, which concrete aggregator types to use, and which dimensions to aggregate by.
+
+## Use Cases
+
+Monitoring and alerting systems commonly use the data provided through metric events or tracers, after applying various aggregations and converting into various exposition format. After getting the data, the systems need to be able to see the data. The StdoutExporter will be used here to print data through Stdout, this is seen as a simple exporter where the user doesn’t have the burden of implementing or setting up a protocol dependent exporter.
```

Comment:
Curious - `tracers` or `traces`?

Reply:
We get spans from tracers, thats why I said tracers here.